Pod/Classes/Terminals/Utility/GlobalMobileSDK/GMSRequestHelper.swift

```diff
@@ -106,30 +106,30 @@ class GMSRequestHelper {
                                                      invoiceNumber: nil,
                                                      operatingUserId: nil)
     }
-    
+
     public static func buildCreditReversalRequest(builder: GMSCreditReversalBuilder) -> Transaction? {
         let total: Decimal? = builder.amount as Decimal?
         let posReferenceNumber: String? = builder.referenceNumber
         let transactionId: String? = builder.transactionId
         let reversalReason: ReversalReason = HpsC2xEnums.reversalReasonCodeToReversalReason(builder.reason)
 
         if let clientTransactionId = builder.clientTransactionId {
-            return ReversalTransaction.reversal(clientTransactionId: clientTransactionId,
+            return ReversalTransaction.reversal(clientTransactionId: "\(clientTransactionId)",
                                                 gatewayTransactionId: transactionId,
                                                 reversalReason: reversalReason,
                                                 posReferenceNumber: posReferenceNumber,
                                                 amount: decimalToUint(total) ?? 0,
                                                 tlv: nil)
         }
 
-        return ReversalTransaction.reversal(clientTransactionId: UUID(),
+        return ReversalTransaction.reversal(clientTransactionId: "",
```

Comment:
The value for `clientTransactionId` should be obtained from the builder when present.